src/FakeItEasy.Analyzer/ArgumentConstraintNullabilityMismatchCodeFixProvider.cs

```diff
@@ -0,0 +1,144 @@
+
+        private static string MakeNotNullConstraintCodeFixTitle =>
+            DiagnosticDefinitions.ResourceManager.GetString(nameof(MakeNotNullConstraintCodeFixTitle));
+
+        public override Task RegisterCodeFixesAsync(CodeFixContext context)
+        {
+            var diagnostic = context.Diagnostics.FirstOrDefault();
+            if (diagnostic == null)
+            {
+                return CompletedTask;
+            }
+
+            context.RegisterCodeFix(
+                CodeAction.Create(
+                    MakeConstraintNullableCodeFixTitle,
+                    ct => this.MakeConstraintNullableAsync(context, diagnostic, ct)),
+                diagnostic);
+
+            context.RegisterCodeFix(
+                CodeAction.Create(
+                    MakeNotNullConstraintCodeFixTitle,
+                    ct => this.MakeNotNullConstraintAsync(context, diagnostic, ct)),
+                diagnostic);
+
+            return CompletedTask;
+        }
+
+        private static MemberAccessExpressionSyntax GetConstraintNode(Diagnostic diagnostic, SyntaxNode root)
+        {
+            // getInnermostNodeForTie: true to disambiguate between ArgumentSyntax and
+            // MemberAccessExpressionSyntax, which have the same source span. We want the
+            // MemberAccessExpressionSyntax node, which is the innermost one.
+            return root.FindNode(diagnostic.Location.SourceSpan, getInnermostNodeForTie: true) as MemberAccessExpressionSyntax;
+        }
+
+        private static MemberAccessExpressionSyntax SimpleMemberAccess(ExpressionSyntax expression, SimpleNameSyntax name)
+        {
+#if CSHARP
+            return SyntaxFactory.MemberAccessExpression(SyntaxKind.SimpleMemberAccessExpression, expression, name);
+#elif VISUAL_BASIC
+            return SyntaxFactory.SimpleMemberAccessExpression(expression, name);
+#endif
+        }
+
+        private async Task<Document> MakeConstraintNullableAsync(CodeFixContext context, Diagnostic diagnostic, CancellationToken cancellationToken)
+        {
+            var root = await context.Document.GetSyntaxRootAsync(cancellationToken);
+
+            var constraintNode = GetConstraintNode(diagnostic, root);
+
+            // The A<T> type
+            var aType = constraintNode?.Expression as GenericNameSyntax;
+
+            // The T type
+            var constraintType = aType?.TypeArgumentList.Arguments.FirstOrDefault();
+            if (constraintType != null)
+            {
+                // The T? type
+                var nullableConstraintType = SyntaxFactory.NullableType(constraintType);
+
+                // Replace T node with T? and return updated document
+                var newRoot = root.ReplaceNode(constraintType, nullableConstraintType);
+                return context.Document.WithSyntaxRoot(newRoot);
+            }
+
+            return context.Document;

```

Comment:
If the node with the diagnostic doesn't match our expectations, we just return the original document. Not sure if this is the correct way to handle it, but I don't think it's possible anyway.